Hardware/Hardware/CardReader/StripeCardReader/PaymentIntent+Stripe.swift

```diff
@@ -32,6 +32,13 @@ protocol StripePaymentIntent {
     var charges: [StripeTerminal.Charge] { get }
 }
 
-
-extension StripeTerminal.PaymentIntent: StripePaymentIntent { }
+/// Our implementation of PaymentIntent does not allow`id` to be nullable, as offline payments are not supported in the app.
+/// This differs from StripeTerminal.PaymentIntent since 3.0.0, these can be nullable to support offline payments.
+/// In order to not make our implementation accept a property that does not correspond to the app logic,
+/// we provide an empty string if no stripeID can be found.
+extension StripeTerminal.PaymentIntent: StripePaymentIntent {
+    var id: String {
+        stripeId ?? ""
+    }
+}
```

Comment:
This is needed for conformance since we have to keep `stripeId` optional in our `StripePaymentIntent` protocol, but `id` non-optional on `PaymentIntent`. If we make the init failable, with a non-nullable `id` then we could remove it, however this causes the problems described on  https://github.com/woocommerce/woocommerce-ios/pull/11080#discussion_r1398596555. 

Happy to iterate if you have a recommendation. I could have missed a different way to handle it.